### PR TITLE
automatically determine if endpoint is KM2 or DLX

### PR DESCRIPTION
**Pull request by dm82m**
@chiefdeputy can you have a look if that works out in your case. any feedback highly appreciated!

### CODE REVIEW

**Comment by dm82m** [comment]
@chiefdeputy I released it as pre-release beta, you can also enable beta in HACS and use 0.1.2b to test it out in your instance

File: custom_components/deltasol/deltasolapi.py [hunk 1]
```diff
@@ -47,31 +46,35 @@ def fetch_data(self):
...
             authId = response[0]['result']['authId']
             
             payload = "[{'id': '1','jsonrpc': '2.0','method': 'dataGetCurrentData','params': {'authId': '" + authId + "'}}]"
-            
+
             response = requests.request("POST", url, headers=headers, data = payload).json()
             _LOGGER.debug(f"KM2 response {response}")
             response = response[0]["result"]
+
+        except:
```
Comment (chiefdeputy) [inline custom_components/deltasol/deltasolapi.py:70]: Perhaps check explicitly for `ValueError` ?
In that case we don't mask other exceptions that might occur for example when doing the post request.

**Review by chiefdeputy — APPROVED** [review]
Works with my DL2.

**Comment by dm82m** [comment]
@chiefdeputy can you test it again with the explicit catch of JSONDecodeError, I updated 0.1.2b so you can just reload it

**Comment by chiefdeputy** [comment]
> @chiefdeputy is it working?

Not as it is right now. Either use ValueError, or upgrade request package and use requests.exceptions.JSONDecodeError or back to the general exception.

**Comment by ostat** [comment]
I am yet to give this a try on our DL3, however what about keeping mode and having "auto[default], km2, dlx". I would prob rather set it than have it try km2 to get an error each time. 
Perhaps on success it could set the config to the correct value?

Just a thought, If the km2 credentials are wrong or it's down it will attempt the dlx and not log a useful message.

**Comment by ostat** [comment]
Submitted an enhanced auto detect #12

**Pull request merged**